src/components/MarketMySharesForm/index.js

```diff
@@ -256,82 +345,26 @@ class MarketMySharesForm extends Component {
+              <button type="button" className="btn btn-link" onClick={this.handleCloseSellView}>
+                Cancel
+              </button>
             </div>
           </div>
           {submitFailed && (
-          <div className="row">
-            <div className="col-md-9 col-md-offset-3 marketMyShares__errorColumn">
-              Sorry - your share sell could not be processed. Please ensure you&quot;re on the right network.
+            <div className="row">
+              <div className="col-md-9 col-md-offset-3 marketMyShares__errorColumn">
+                Sorry - your share sell could not be processed. Please ensure you&quot;re on the right network.
+              </div>
             </div>
-          </div>)}
+          )}
         </form>
       </div>
     )
   }
 
-  generateTableRows() {
-    const tableRows = []
-
-    const { marketShares, market } = this.props
-    const { extendedSellId } = this.state
-
-    const resolved = market.oracle.isOutcomeSet || market.event.isWinningOutcomeSet
-
-    marketShares.forEach((share, shareIndex) => {
-      const probability = calcLMSRMarginalPrice({
-        netOutcomeTokensSold: market.netOutcomeTokensSold.slice(0),
-        funding: market.funding,
-        outcomeTokenIndex: share.outcomeToken.index,
-      })
-      const maximumWin = calcLMSROutcomeTokenCount({
-        netOutcomeTokensSold: market.netOutcomeTokensSold.slice(0),
-        funding: market.funding,
-        outcomeTokenIndex: share.outcomeToken.index,
-        cost: share.balance,
-      })
-
-      tableRows.push((
-        <tr className="marketMyShares__share" key={share.id}>
-          <td>
-            <div
-              className={'shareOutcome__color'} style={{ backgroundColor: COLOR_SCHEME_DEFAULT[share.outcomeToken.index] }}
-            />
-          </td>
-          <td className="">
-            {getOutcomeName(market, share.outcomeToken.index)}
-          </td>
-          <td>
-            {Decimal(share.balance).div(1e18).gte(LOWEST_DISPLAYED_VALUE) ?
-              <DecimalValue value={Decimal(share.balance).div(1e18)} /> : `< ${LOWEST_DISPLAYED_VALUE}`}
-          </td>
-          <td>
-            <DecimalValue value={maximumWin.mul(probability).div(1e18)} />&nbsp;
-            <CurrencyName collateralToken={market.event.collateralToken} />
-          </td>
-          <td>
-            {/* eslint-disable no-script-url */}
-            {!resolved && <a href="javascript:void(0);" className="marketMyShares__sellButton" onClick={e => this.handleShowSellView(e, share.id)}>Sell</a>}
-            {/* eslint-enable no-script-url */}
-          </td>
-        </tr>
-      ))
-
-      if (share.id === extendedSellId) {
-        tableRows.push((
-          <tr className="marketMyShares__sellView" key={`${share.id}__sell`}>
-            <td colSpan={5}>
-              {this.renderSellShareView()}
-            </td>
-          </tr>
-        ))
-      }
-    })
-
-    return tableRows
-  }
-
   render() {
```

Comment:
Refactor this render method creating private components for each situation. Having more than one return in the render method is confusing.
```
const hasShares = !marketShares || !marketShares.length;
return hasShares ? <SharesMarket /> : <EmptyMarket />
```
(can't find better names :) )

When I say Private components I mean something like: 
```
const EmptyMarket = () => <div ...>
    ....
</div>

//SharesMaker I would create this component as a functional component in ./SharesMarket otherwise this component is huge 400 lines is too much.

class MarketMySharesForm extends Component { 
```